src/dev/build/tasks/os_packages/package_scripts/post_install.sh

```diff
@@ -4,36 +4,56 @@ set -e
 case $1 in
   # Debian
   configure)
-    if ! getent group "<%= group %>" >/dev/null; then
+    if ! getent group "<%= group %>" >/dev/null 2>&1 ; then
+      echo -n "Creating <%= group %> group..."
       addgroup --quiet --system "<%= group %>"
+      echo " OK"
     fi
 
-    if ! getent passwd "<%= user %>" >/dev/null; then
-      adduser --quiet --system --no-create-home --disabled-password \
-      --ingroup "<%= group %>" --shell /bin/false "<%= user %>"
+    if ! id "<%= user %>" >/dev/null 2>&1 ; then
+      echo -n "Creating <%= user %> user..."
+      adduser --quiet \
+              --system \
+              --no-create-home \
+              --home /nonexistent \
+              --ingroup "<%= group %>" \
+              --disabled-password \
+              --shell /bin/false \
+              "<%= user %>"
+      echo " OK"
     fi
   ;;
   abort-deconfigure|abort-upgrade|abort-remove)
   ;;
 
   # Red Hat
   1|2)
-    if ! getent group "<%= group %>" >/dev/null; then
+    if ! getent group "<%= group %>" >/dev/null 2>&1 ; then
+      echo -n "Creating <%= group %> group..."
       groupadd -r "<%= group %>"
+      echo " OK"
     fi
 
-    if ! getent passwd "<%= user %>" >/dev/null; then
-      useradd -r -g "<%= group %>" -M -s /sbin/nologin \
-      -c "kibana service user" "<%= user %>"
+    if ! id "<%= user %>" >/dev/null 2>&1 ; then
+      echo -n "Creating <%= user %> user..."
+      useradd --system \
+              --no-create-home \
+              --home-dir /nonexistent \
+              --gid "<%= group %>" \
+              --shell /sbin/nologin \
+              --comment "kibana service user" \
+              "<%= user %>" \
+      echo " OK"
     fi
   ;;
 
   *)
-      echo "post install script called with unknown argument \`$1'" >&2
-      exit 1
+    echo "post install script called with unknown argument \`$1'" >&2
+    exit 1
   ;;
 esac
 
-chown -R <%= user %>:<%= group %> <%= optimizeDir %>
-chown <%= user %>:<%= group %> <%= dataDir %>
+chown <%= user %>:<%= group %> <%= optimizeDir %>
 chown <%= user %>:<%= group %> <%= pluginsDir %>
+chown -R <%= user %>:<%= group %> <%= configDir %>
```

Comment:
I have some doubts here as `configDir` (`/etc/kibana`) directory is usually owned by root and there is only its content which is owned by kibana user and kibana group. However I don't know if this can be a problem or not.

I am not totally satisfied with the proposed fix as this is essentially using a big hammer to change recursively ownership of all existing files. I checked on some of my kibana installs and it seems that there is at most hundred files at least on those instances.

I wanted to keep the changeset small but maybe we could have a more chirurgical approach by looking up in these directories if there is files having only a uid/gid without an associated user/group. Having this knowledge would allow to:
- only change ownership of files if needed (but it should be an all or nothing change)
- reuse uid and gid previously assigned to kibana user and group to avoid "waste" of uid/gid on the system.

What do you think about this ?

Reply:
Thanks @arenard, I'm leaning towards the reuse portion, if only to be consistent with elasticsearch.  It looks like they did chown until [https://github.com/elastic/elasticsearch/pull/37242/files](recently).  We have a parallel issue filed at https://github.com/elastic/kibana/issues/28690.

It looks like both work, and honestly I'm mostly indifferent. GID recovery does seem to be the right way, combined with consistency w/ elasticsearch.  I'd have to dig to see what that looks like in our environment, not sure.  Thoughts?